Update some mappings (Part 14)

JLHwung opened the pull request:
This PR includes all fixes applied to my fork according to the Unicode 16 IDS. I planned to submit them after https://github.com/Jackchows/Cangjie5/pull/248 is merged but apparently I forgot to do so.

For reviewers: The `Cangjie5_HK.txt` are missing a few updates because they were made before this file is available. The `Cangjie5.txt` and `_SC`, `_TC` are correctly synchronized (hopefully). It will be great if we can somehow sync `Cangjie5.txt` to `Cangjie5_HK.txt`. I checked the `README` documents in the `scripts` folder but couldn't figure out how to do so. Helps are appreciated.

Fixes #314 
Fixes #315
Fixes #316

JLHwung committed:
mark 𰔚	qbsi as [u]

JLHwung committed:
𥉬	bumnf

Also mark 𥉬	bnf/xbnf as [U]

we have

𪶭	emnf
𤍅	fmnf
𦤭	hkmnf
𠗹	immnf
𤹐	kmnf
𧏲	limnf
𩄰	mbmnf
𬟜	mftja
烈	mnf
𢟏	pmnf
𰔔	qmnf
𨃻	rmmnf
𠺅	rmnf
𫊔	tamnf
𦶣	tmnf
𦃾	vfmnf
𥉬	xbnf
𠺅	xrmnf

JLHwung committed:
𫱂	vmue

we have

涵	emue
𭡝	qmue
𭡝	xqmue
𬢲	yrmue

JLHwung committed:
Add G/UTC source mapping for 𧊥

G/UTC source has ⿰虫戍

JLHwung committed:
mark 𤶊	koyi as [U]

JLHwung committed:
𧈱	iplmi

JLHwung committed:
mark 𤆥	fpim as [u]

JLHwung committed:
mark 𧦠	yrmru as [U]

JLHwung committed:
𣹶	eooe

we have

夎	oghe
𠹐	rooe
蓌	tooe

JLHwung committed:
mark 𩣙	sfmjl as [U]

JLHwung committed:
mark 𬙐	ounme as [u]

JLHwung committed:
𬃤	dhxs

we have

𣉃	ahxs
𨩤	chxs
𫃈	fdhxs
𢰻	qhxs
𦖳	sjhxs
𡞲	vhxs
𡞲	xvhxs

JLHwung committed:
𬈍	ehxs

we have

𣉃	ahxs
𨩤	chxs
𫃈	fdhxs
𢰻	qhxs
𦖳	sjhxs
𡞲	vhxs
𡞲	xvhxs

JLHwung committed:
𥜞	mfhxs

we have

𣉃	ahxs
𨩤	chxs
𫃈	fdhxs
𢰻	qhxs
𦖳	sjhxs
𡞲	vhxs
𡞲	xvhxs

JLHwung committed:
remove 𬌫	khvi

we have

𬁲	bhi
𥃼	buhi
𣏈	dhi
𭯷	ehi
𥝠	hdhi
𠂓	hhi
么	hi

JLHwung committed:
mark 𢮚	qahk as [u]

JLHwung committed:
𣽷	eboh

we have

赐	boaph
𰂭	oboh

also removed 𣽷	xxebc because eboh does not colide with any mappings.

JLHwung committed:
𮩶	sfvo

Also mark 𮩶	sfno as [u]

we have

𣲄	evo
𥤳	jcvo
𤱑	jwvo
𰞆	lpvo
𰀬	nsvo
乆	vo

JLHwung committed:
𭴆	yvf

JLHwung committed:
𮌳	gkb

𮌳	xgkbl is removed because we already have 𰉦	xgkb

JLHwung committed:
𢅪	lbifp

we have

慼	ifp
𧞰	lifp
𭟍	pifp
𢷾	qifp
𡂔	rifp
慼	xifp

JLHwung committed:
𥀽	geifp

we have

慼	ifp
𧞰	lifp
𭟍	pifp
𢷾	qifp
𡂔	rifp
慼	xifp

JLHwung committed:
𥍚	buvfk

we have

𨰺	cvfk
𤅶	evfk
𤓩	fvfk
𡔖	gvfk
𥘂	mrvfk
𫶦	uvfk
變	vfok

JLHwung committed:
remove incorrect [u] mark

JLHwung committed:
𰢍	icmgi

we have

𰢍	idmgi [u]

but we don't have a mapping for the code chart glyph

JLHwung committed:
Add G-Source mapping for 𢡍

Also mark 𢡍 as [U]

JLHwung committed:
𮝛	jjrvb

we have

𦝖	brvb
𣷢	ervb
𥺳	fdrvb
𥟮	hdrvb
𥮪	hrvb
𥚝	ifrvb
𤟕	khrvb
𥚝	mfrvb
𨡠	mwrvb
𧤉	nbrvb
𩸠	nfrvb
𢯇	qrvb
𦙃	rvb
𦲭	trvb
𦂝	vfrvb
𦲭	xtrvb
𧩑	yrrvb

JLHwung committed:
𠵭	rrvb

we have

𦝖	brvb
𣷢	ervb
𥺳	fdrvb
𥟮	hdrvb
𥮪	hrvb
𥚝	ifrvb
𤟕	khrvb
𥚝	mfrvb
𨡠	mwrvb
𧤉	nbrvb
𩸠	nfrvb
𢯇	qrvb
𦙃	rvb
𦲭	trvb
𦂝	vfrvb
𦲭	xtrvb
𧩑	yrrvb

JLHwung committed:
𭟶	rbip

we have

𦙃	rvb
𠦣	rbj

JLHwung committed:
A V-Source mapping for 𢬭

Also mark 𢬭	qiei as [U]

JLHwung committed:
Add G-Source mapping for 𩒘

JLHwung committed:
Add G-Source mapping for 𢫻

We already have qmgi for ⿰扌丟

now adding qhgi for ⿰扌丢

JLHwung committed:
𡏬	eug

G4K glyph gives eug. Also marks evg as [u]

JLHwung committed:
Add G-Source mapping for 𢪾

G-Source glyph has ⿰扌罒

JLHwung committed:
𢈅	ihvo

we have

坬	ghvo
孤	ndhvo
弧	nhvo
呱	rhvo

JLHwung committed:
mark 𤷭	kham as [u]

JLHwung committed:
𡹴	uhdt

we have

乗      hdt

JLHwung committed:
mark 𦝂	bham as [u]

JLHwung committed:
Add G-Source mapping for 𢚇

G-source has ⿰忄㞷

JLHwung committed:
𢕏	howtj

we have

畢	wtj

JLHwung committed:
mark 𮨰	hnjbc as [u]

JLHwung committed:
𨸲	nlylb

we have

市	ylb

JLHwung committed:
𥢴	hdmbl	[u]

we have

零	mboml

JLHwung committed:
𭽣	dehpm

we have

氐	hpm
氐	hvpi

JLHwung committed:
Add G-Source mapping for 𤝵

G-source has ⿰犭𡉚

JLHwung committed:
𦌻	ywlc

Mark 𦌻	wlyrc as [u], which is T-Source glyph prior to Unicode 6.0

JLHwung committed:
mark 𮚓	bcnme as [u]

JLHwung committed:
mark 𣇝	aiml as [u]

JLHwung committed:
mark 𬙲	tqmt as [u]

JLHwung committed:
𡞍	vjwi

we have

叀	jwi

JLHwung committed:
remove 𢸤	qtgi

we have

𢸤	qtgs

JLHwung committed:
𫫗	rmfu

we have

覔 mfbuu

JLHwung committed:
Add T-Source mapping for 䝈

T-Source has ⿰豕厄 and we have

厄	msu

JLHwung committed:
𰕲	lha

we have

帰	lhsmb

JLHwung committed:
归	lism

http://chidic.eduhk.hk/v.php?dicword=%E5%BD%92

JLHwung committed:
帰	lismb

we have

归 lism [O]

JLHwung committed:
𧊎	liiv

we have

𧵝      bciv
眓      buiv
鉞      civ
𪌝      deiv
𣐋      div
𩎙      dqiv
泧      eiv
越      goiv
𪀚      hfij
𩿠      hfiv
魆      hiiv
䬂      hniv
𬱸      hniv
𱥟      huiv
戉      iv

JLHwung committed:
𭹠 mgbhg

we have

㸒      bhg
㸒      bmg
𪾮      bubhg
𣓆      dbhg
淫      ebhg
𭵂      fbhg
𭵂      fbmg
𡍛      gbhg
𥮍      hbhg
𠗝      imbhg
𤷏      kbhg
𭹠      mgbmg
𣨮      mnbhg
𥓕      mrbhg
𭀩      mubhg
𨺄      nlbhg
𨺄      nlbmg
𠉰      obhg
𱓈      rbhg
𱓈      rbmg
𨂘      rmbhg
𦲻      tbmg
婬      vbhg
𦁍      vfbhg
𦁍      vfbmg
㸒      xbmg
𨓮      xybhg
𨓮      ybhg

Also marked mgbmg as [u] since there is no T glyph.

JLHwung committed:
𪖩	hlmbt

we have

𡨴      jmbn
𱮖      khnit
𤶰      kmbt
𮕜      mbt
𮕜      xmbt

JLHwung committed:
𱥻	eshc/eshi

we have

䘐      htshc
𠟐      orshc
𠟐      orshi
𠛂      scln
刅      shc
刅      shii
剙      ttshc   [O]
刱      ttshc
刱      ttshi
剙      ttshi   [O]
𥾠      vfshc
𠟐      xorsh
刱      xttsh
𪲠      xxmid
剙      xyjyj   [Ou]
剙      yjyjc   [Ou]
剙      yjyji   [Ou]
𬔚      ytshc

JLHwung committed:
Add V-Source mapping for 𮈩

JLHwung committed:
𣸶	emnj

we have

𱣴      dmnj
𱣴      dmnq    [u]
𱸐      hmnj
𰤽      mjbt
残      mnij
残      mniq
𰤽      xxmjb

JLHwung committed:
𩳱	hihxl

we have

𦥔 hxl

JLHwung committed:
𮀌	mrvo

we have

𣲄      evo
𥤳      jcvo
𤱑      jwvo
𰞆      lpvo
𮀌      mrno
𱳸      mrvoj
𰀬      nsvo
𢱿      qibo
𮩶      sfno    [u]
𮩶      sfvo
𠔨      vicmc   [u]
𠔨      vihhc   [u]
𤯖      vihqm
𠔨      vitxc
乆      vo

JLHwung committed:
𡠃	jvv

we have

戉	iv

JLHwung committed:
𡒅	giqw

we have

𢊂      iqmw

JLHwung committed:
𡍙	gthv

we have

𱣩      dthu
𲩛      ethu
𱫖      fthu

𥟌      hdthn
𱿬      lithu
𬵘      nfthu
𱠍      qthu
𱓄      rthu
𱽐      thnv
𱡵      tuyhs
𱟤      tvhss   [u]
𱟤      tviss
𲸂      vfthu
𫿕      ysoku
𫿕      ysokv
𱡴      ysthu

JLHwung committed:
𥟌	hdthu

The V glyph is changed from ⿰禾芃 to ⿰禾𱽐 in Unicode 17.

Also mark 𥟌	hdthn as [U]

JLHwung committed:
𪱡	bomd

The V glyph is changed from ⿰月佘 to ⿰月余 in Unicode 17.

Also mark 𪱡	bomf as [U]

JLHwung committed:
𣛬	dhhl

we have

𬁋      ahhl
卿      hhail
卿      hhhpl   [u]
卿      hhhsl
𤫈      mghlp
𮬐      nfhhl
𦺄      thhl
𲛤      vhhl
𬁋      xahhl
𮬐      xnfhh
𦺄      xthhl
𲛤      xvhhl

Also removing 𣛬 xdhll as both xdhhl and xxdhh have been occupied

JLHwung committed:
𭩋	bjmo

we have

𥋰      bujmo
𤀋      ejmo
𤃚      enio
𡩟      jhxo
𡩟      jmho

JLHwung committed:
Add G-Source mapping for 𡽃

The G-Source glyph gives ⿰阝⿱𢀡山.

JLHwung committed:
𭵲	flmi

we have

𨪙      clmi
𭵲      fvmi
𬌒      libm
𰊈      lig
奨      lik
螀      lilmi
醤      limcw
𫦋      lish
𮔬      livmi
将      lmbdi
将      lmnii
鳉      nmlmi
锵      oplmi
𬧀      rmlmi
蒋      tlmi
𡺃      ulmi

JLHwung committed:
𫽣	qlmi

we have

𨪙      clmi
𭵲      fvmi
𬌒      libm
𰊈      lig
奨      lik
螀      lilmi
醤      limcw
𫦋      lish
𮔬      livmi
将      lmbdi
将      lmnii
鳉      nmlmi
锵      oplmi
𬧀      rmlmi
蒋      tlmi
𡺃      ulmi

JLHwung committed:
𣇭	ahom

we have

𭰵      ehom
𥯁      hhom
徃      hohqm
𱻃      wlhom
𲁬      yrhom

Also removed 𣇭      xahog because ahom is currently unique.

JLHwung committed:
𣶬	enem

we have

𣷉      emem
亟G      nem

The right component follows G source convention. Since 𣷉 and 𣶬 are separately encoded, we don't have to add emem for 𣶬.

JLHwung committed:
Add J-Source mapping for 𢔔

The J-source glyph is included in Unicode 16.0

Jackchows commented on Cangjie5.txt:
```diff
@@ -38770,7 +38777,6 @@
 𭈝	ilr
 𢉫	ilsl
 归	ilsm
-帰	ilsmb
```
It's better to keep `il` to match `班`.

JLHwung replied:
Understood, the 𬼀 part is in such a mess. Even 漢文庫典 does not treat 班 and 帰 consistently. I will review the 𬼀 changes more carefully.

Jackchows commented on Cangjie5.txt:
```diff
@@ -74791,6 +74804,7 @@
 𭟙	pmbg
 愞	pmbk
 𲟹	pmbl
+𢡍	pmbl	[U]
```
I think it's not necessary to add an old version of the graph. If it already exists, we can keep it, but not add a new one.

JLHwung replied:
Makes sense, will update.

Jackchows commented on Cangjie5.txt:
```diff
@@ -77317,6 +77333,7 @@
 𫾚	qidr
 攠	qidy
 𰛣	qie
+𢬭	qiei	[U]
```
I think it's not necessary to add an old version of the graph. If it already exists, we can keep it, but not add a new one.

Jackchows commented on Cangjie5.txt:
```diff
@@ -76051,7 +76066,7 @@
 𪃣	qahaf
 𱈻	qahaf
-𢮚	qahk
+𢮚	qahk	[u]
```
It's ok to delete `qahk`.

Jackchows commented on Cangjie5.txt:
```diff
@@ -46452,6 +46458,7 @@
 𤣖	khujt
 𤻖	khul
 猯	khumb
+𤝵	khumg
```
Also, mark `khuqm` as `[U]`.

Jackchows commented on Cangjie5.txt:
```diff
@@ -23019,10 +23023,10 @@
 𱕞	gehyr	[u]
 𮰚	gei
 墚	geid
+𥀽	geifp
```
I think it's better to mark `geifp` as `[u]`.

JLHwung replied:
This change was introduced in https://github.com/JLHwung/Cangjie5/commit/e95d2e5ea904df953afa5203ba4a77d6e0e72009, where we changed `geihp` to `geifp` to align with other 慼 characters. So currently `geifp` is the only Cangjie code of 𥀽 and it would be very confusing if the only code is marked as `[u]`. Did you mean marking the old code `geihp` as `[u]`?

Jackchows replied:
I mean
```
𥀽  geihp
𥀽  geifp   [u]
```

JLHwung replied:
Oh I see. The bottom component of is 𥀽 is 慼T (ihp?). Well in that case I agree that `geihp` is more appropriate, and we should probably also add `ihp` for 慼T. I thought `ifp` is good for 慼T because `ifp` is provided in [漢文庫典](https://chidic.eduhk.hk/v.php?dicword=%E6%85%BC), but apparently that glyph only matches glyphs from non-T sources.

Jackchows commented on Cangjie5.txt:
```diff
@@ -104143,7 +104163,7 @@
 詈	wlymr
 𦋾	wlypm
-𦌻	wlyrc
+𦌻	wlyrc	[u]
```
The J-source graph is `wlyrc`.

Jackchows commented on Cangjie5.txt:
```diff
@@ -127871,7 +127886,7 @@
 謌	yrmrr
 䛠	yrmrt
-𧦠	yrmru
+𧦠	yrmru	[U]
```
The J-source graph is `yrmru`.

Jackchows commented on Cangjie5_HK.txt:
```diff
@@ -30804,10 +30806,10 @@
 簣	hlmc
 𣃐	hlmcs
 𪗁	hlmhm
-𪖩	hlmht
```
The G-source and T-source graphs are `hlmht`, and the J -source graph is `hlmbt`.

JLHwung replied:
Good catch. This is a false positive from the IDS report as the IDS does not differentiate between ⿱一血 and 𮕜. Will revert this change.

Jackchows commented on Cangjie5.txt:
```diff
@@ -23945,6 +23949,7 @@
 𣊓	gka
 𪢹	gkb
 𰉦	gkb
+𮌳	gkb
```
Also, remove `gkbl`.

Jackchows commented on Cangjie5.txt:
```diff
@@ -86038,7 +86056,7 @@
 𩥩	sfnmm
 𩦑	sfnmr
-𮩶	sfno
+𮩶	sfno	[u]
```
`sfno` is a mistake, so just remove it.

JLHwung replied:
The `sfno` is to map the unifiable unencoded form ⿰馬久 to 𮩶. I think it is okay to keep `[u]` since 久 is the modern normalized form of 乆.

Jackchows replied:
I’m not sure whether ⿰馬久 can be unified with 𮩶, since 汣/𣲄, 畞/𤱑, and 𥤯/𥤳 have different code points.

JLHwung replied:
⿰馬久 would be unified with 𮩶 if it were submitted today according to the [UCV #428](https://hc.jsecs.org/irg/iwds/ucv?show=428#entry-428), unless ⿰馬久 is not cognate with 𮩶. Of course the UCV can not apply to previously disunified characters, but it would weight in submitted characters in the future.

/cc @eisoch

JLHwung replied:
Removed `sfno`.

Jackchows replied:
Thank you for the explanation. This document has been very helpful.

Jackchows commented on Cangjie5.txt:
```diff
@@ -30692,7 +30698,6 @@
 𥫯	hl
 𣊅	hla
 𣊆	hla
-𰕲	hla
```
`hla` is correct.

JLHwung replied:
OK, reverted this change. In sum 𰕲 now has two mapping, one is `lha` to align with 帰, and `hla` to (sort of) align with 班.

Jackchows reviewed (commented):
Thank you for your help! I’ve left a some comments with my thoughts—please take a look when you have time.

It’s enough to update only `Cangjie5.txt`. I’ll update my database, and `_TC`, `_HK` and `_SC` are automatically generated by the program. It’s also unnecessary to edit the codes with the ‘x’ prefix, since they’re generated by the program as well.

JLHwung committed:
address review comments

JLHwung commented:
Thank you very much for such a detailed and helpful review comments. I have made most changes and resolved related comments.

JLHwung committed:
𧇕	ypvmw

we have

鍿      cvmw
湽      evmw
輺      jjvmw
𦿨      thdw
葘      tvmw
𱹼      vfvmw
𱰭      vmw

JLHwung committed:
Add J-source mapping for 𤄴

we have

𧹎      bcgcc
𧸇      gcwbc
𥩐      jcgcc
𧟎      lgcc
𨙜      ygcc

JLHwung committed:
Add T-source mapping for 慼

慼	ihp

Also handled other characters with the 慼 component.

JLHwung committed:
𬘽	vmsyj

we have

𣹲      esyj
𡎰      gsyj
稺      hdsyj
𢕌      hosyj
𧎨      lisyj
𧛺      lsyj
搱      qsyj
𠞂      sjln
屖      sytj
𦃘      vfsyj
𡟭      vsyj
𡎰      xgsyj
𠞂      xsjln
謘      yrsyj
遟      ysyj

JLHwung committed:
remove 𮩶	sfno

Jackchows reviewed (commented):
Thanks for your submission.
